chapters/en/slides/module7/module7_02.Rmd

```diff
@@ -0,0 +1,342 @@
+choropleth = (alt.Chart(state_map).mark_geoshape().transform_lookup(
+    lookup='id',
+    from_=alt.LookupData(state_pop, 'id', ['number_of_asthma_cases', 'state']))
+.encode(
+    color=alt.Color('number_of_asthma_cases:Q', title='Asthma cases'),
+    opacity=alt.condition(map_hover, alt.value(1), alt.value(0.2)),
+    tooltip=['state:N', alt.Tooltip('number_of_asthma_cases:Q', title='Asthma cases')])
+.add_selection(map_hover)
+.project(type='albersUsa')
+.properties(height=150, width=300))
+choropleth
+```
+
+Notes:
+Selections work with all types of charts in Altair.
+In this slide we have recreated one of the maps we made in module 6,
+showing the number of asthma cases per state as a choropleth.
+
+We have added a helpful tooltip
+that shows which state we are hovering over
+as well as the exact number of asthma cases in that state,
+which would be hard to read out from the color alone.
+
+We used `alt.Tooltip` to change the title in the tooltip,
+just as when we change the title of an axis or legend.
+
+We also added a `selction_single` interaction,
+but instead of binding it to mouse clicks (the default),
+we bound it to `'mouseover'`,
+which is when we are hovering over a state with the mouse pointer.
+
+In this slide we can use this interaction
+to highlight a particular state in the choropleth
+and in the next slide
+we will see how we can also link it to another chart.
+
+We set the dimensions of the plot to fit on the slide.
+
+---
+
+## Linking a map selection to another chart
+
+```{python}
+points = (alt.Chart(state_pop).mark_circle(size=70).encode(
+    alt.X('asthma_cases_per_capita', title='Asthma cases / capita', scale=alt.Scale(zero=False)),
+    alt.Y('number_of_asthma_cases', title='Asthma cases'),
+    stroke=alt.condition(map_hover, alt.value('black'), alt.value('#ffffff')),
+    color='number_of_asthma_cases')).properties(height=150, width=300)
+choropleth & points
+```
+
+Notes:
+To link an interaction on a map to another chart,
+we use the same logic as we did in previous slides
+for linking the barchart and scatterplot.
+
+Here we create a separate scatterplot
+where the stroke color of each points depends on
+whether we have selected that state on the map or not.
+
+By default
+all states on the map are selected 
+so all the points have a black outline.
+As we hover over a state in the map,
+the outline changes to white for the points
+whose states are not selected in the map.
+
+In addition to adding one more visualization for the asthma cases,
+using the map to identify states can be more helpful
+than just adding a tooltip to the scatterplot
+since we might already have an intuition 
+for where different states are located on the map
+and don't have to go through each of the point's tooltip
+looking for the state name of interest.
+
+TODO we could change this to be another variable in the map like population, I don't have a strong opinion, both have advantages.
+
```

Comment:
This slide kind of begs the linked view to also go in the other direction - from the scatter plot to the map. For example, it would be cool to have the state name show up when hovering over a point, and then have that state highlighted on the map. Also, it would be cool to be able to select, or brush over multiple points and see them highlighted on the map.

Reply:
adding to wishlist